jmx-scraper/src/integrationTest/java/io/opentelemetry/contrib/jmxscraper/target_systems/MetricsVerifier.java

```diff
@@ -37,113 +32,149 @@ private MetricsVerifier() {}
    * allowingExtraMetrics() method.
    *
    * @return new instance of MetricsVerifier
-   * @see #allowExtraMetrics()
+   * @see #disableStrictMode()
    */
   public static MetricsVerifier create() {
     return new MetricsVerifier();
   }
 
-  @SuppressWarnings("CanIgnoreReturnValueSuggester")
-  public MetricsVerifier allowExtraMetrics() {
+  @CanIgnoreReturnValue
+  public MetricsVerifier disableStrictMode() {
     strictMode = false;
     return this;
   }
 
-  @SuppressWarnings("CanIgnoreReturnValueSuggester")
-  public MetricsVerifier assertGauge(String metricName, String description, String unit) {
+  @CanIgnoreReturnValue
+  public MetricsVerifier register(String metricName, Consumer<MetricAssert> assertion) {
     assertions.put(
         metricName,
         metric -> {
-          assertDescription(metric, description);
-          assertUnit(metric, unit);
-          assertMetricWithGauge(metric);
-          assertThat(metric.getGauge().getDataPointsList())
-              .satisfiesExactly(point -> assertThat(point.getAttributesList()).isEmpty());
+          MetricAssert metricAssert = assertThat(metric);
+          assertion.accept(metricAssert);
+          if (strictMode) {
+            metricAssert.validateAssertions();
+          }
         });
-
     return this;
   }
 
-  @SuppressWarnings("CanIgnoreReturnValueSuggester")
+  // TODO: can now be inlined
+  @CanIgnoreReturnValue
+  public MetricsVerifier assertGauge(String metricName, String description, String unit) {
+    return register(
```

Comment:
[for reviewer] all existing methods can now be inlined by the IDE, we can do that in a follow-up PR to keep this one small.